src/averageAges.js

```diff
@@ -37,7 +40,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenFilter = withChildren ? people.filter(person =>
+    person.sex === 'f' && people.some(child => child.mother === person.name))
```

Comment:
Similar to the previous comment, the condition inside the filter is complex and could be moved to a variable, such as `isMotherWithChild`, to improve code clarity.